### PR TITLE
Added call to defer_to_manifest in before_run step for compile / docs…

### PR DESCRIPTION
**Pull request by seub**
… generate

resolves #

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Comment by ChenyuLInx** [comment]
@seub Thanks for putting up the PR! Do you feel comfortable trying to add some tests for this usecase?
@jtcohen6  looks like this one will generate the docs based on the combination of deferred manifest and current project, is this a feature we want to add.

**Comment by jtcohen6** [comment]
I think the implementation here is based on my comment in https://github.com/dbt-labs/dbt-core/issues/6124#issuecomment-1296368442 - good to go from my point of view

**Comment by seub** [comment]
Thanks @ChenyuLInx and @jtcohen6. I was excited to contribute to dbt but after trying to follow all the steps of the contributing guide I think I got discouraged by all the failing tests (I think it was `make integration` on my end?). I don't think these failures have anything to do with the diff I introduced and I didn't want to spend more time investigating. I also would need to invest more time trying to understand how to add new tests.

Long story short, I appreciate the incitement to contribute, and I tried, but the entry ticket is a bit steep for me. I'll have to leave it to your expert hands and maybe try again to contribute another time.

**Comment by jtcohen6** [comment]
Thanks for taking a stab at it @seub! Closing in favor of the newer PR: https://github.com/dbt-labs/dbt-core/pull/6488
